Update docs (#15)
vtex-apps/bazaarvoice: README.md

| :-----: | :-----------------: | :-------------: | :-------------------: | :---------: | :-----------------: |
|  [1.x]  | **Current Release** |   2019-06-14    |                       |             |         2.x         |

See our [LTS policy](https://github.com/vtex-apps/awesome-io#lts-policy) for more information.

## Table of Contents

- [Usage](#usage)
  - [Blocks API](#blocks-api)
    - [Configuration](#configuration)
  - [Styles API](#styles-api)
    - [CSS namespaces](#css-namespaces)
- [Troubleshooting](#troubleshooting)
- [Contributing](#contributing)
- [Tests](#tests)

## Usage

This app uses our store builder with the blocks architecture. To know more about Store Builder [click here.](https://help.vtex.com/en/tutorial/understanding-storebuilder-and-stylesbuilder#structuring-and-configuring-our-store-with-object-object)
Since this app is an integration app, you should also add it to your store through the `apps` section on the admin menu.

To configure or customize this app you need to do it through the `apps` section of the admin menu and you need to import it in your dependencies in `manifest.json`.

```json
  dependencies: {
    "vtex.bazaarvoice": "1.x"
  }

> ```

Reviewer: Remove text from here on up.

> Then, add the bazaarvoice blocks into your theme as you would with any other block. The bazaarvoice blocks are as it follows:

Reviewer: Change it to instruct to add the the product reviews interfaces, not the bazaarvoice interfaces.

Kind of what is in here: https://github.com/vtex-apps/product-review-interfaces/tree/master/example

- `product-review-form.bazaarvoice` can be used on the [store.product](https://github.com/vtex-apps/store/blob/master/store/interfaces.json#L35)
- `product-reviews.bazaarvoice` can be used on the [store.product](https://github.com/vtex-apps/store/blob/master/store/interfaces.json#L35)
- `product-rating-summary.bazaarvoice` can be used on the [store.product](https://github.com/vtex-apps/store/blob/master/store/interfaces.json#L35)
- `product-rating-inline.bazaarvoice` can be used on the [product-summary](https://github.com/vtex-apps/product-summary/blob/master/store/interfaces.json) and can be used on the [store.product](https://github.com/vtex-apps/store/blob/master/store/interfaces.json#L35)

> Now, you can change the behavior of the bazaarvoice blocks that are in the store. See an example of how to configure:

Reviewer: This should be set in the CMS, right?

```json
"product-reviews.bazaarvoice": {
  "props": {
    "quantityPerPage": 20,
    "quantityFirstPage": 5
  }
}
```

### Blocks API

> When implementing this app as a block, various inner blocks may be available. The following interface lists the available blocks within the bazaarvoice blocks and describes if they are required or optional.

Reviewer: Remove this section, it's not useful

```json
"product-review-form.bazaarvoice": {
  "component": "ReviewForm"
},
"product-reviews.bazaarvoice": {
  "component": "Reviews"
},
"product-rating-summary.bazaarvoice": {
  "component": "RatingSummary"
},
"product-rating-inline.bazaarvoice": {
  "component": "RatingInline"
}
```

These blocks have no required or allowed block. So, any bazaarvoice block implementation do not need any block inside of menu.

#### Configuration

Through the `apps` section, you can change the behaviour of some bazaarvoice components. However, you also can make in your theme app, as Store theme does. The props of the `product-reviews.bazaarvoice` are as it follows:

> 

Reviewer: Instruct to use the CMS instead of setting this in the theme app.

| Prop name      | Type     | Description                                          | Default value |
| -------------- | -------- | ---------------------------------------------------- | ------------- |
| `quantityPerPage`         | `number` | the number of reviews that a review page will have                                            | `10`           |
| `quantityFirstPage`         | `number` | the number of reviews that the first review page will have                                            | Same as `quantityPerPage`          |

The other blocks don't have props.

### Styles API

This app provides some CSS classes as an API for style customization.

To use this CSS API, you must add the `styles` builder and create an app styling CSS file.

1. Add the `styles` builder to your `manifest.json`:

```json
  "builders": {
    "styles": "1.x"
  }
```

> 2. Create a file called `vtex.menu.css` inside the `styles/css` folder. Add your custom styles:

Reviewer: ```suggestion
2. Create a file called `vtex.bazaarvoice.css` inside the `styles/css` folder. Add your custom styles:
```

| reviewsContainerWriteButton | Reviews container write button  | [ReviewsContainer](https://github.com/vtex-apps/bazaarvoice/blob/master/react/components/ReviewsContainer.tsx)  |
| noReviewsContainer | Container that appears when there are no reviews  | [NoReviews](https://github.com/vtex-apps/bazaarvoice/blob/master/react/components/NoReviews.tsx)  |
| writeReviewButton |  Write review button | [ReviewsContainer](https://github.com/vtex-apps/bazaarvoice/blob/master/react/components/ReviewsContainer.tsx),  [NoReviews](https://github.com/vtex-apps/bazaarvoice/blob/master/react/components/NoReviews.tsx) |
| activeStar | Active star  | [Stars](https://github.com/vtex-apps/bazaarvoice/blob/master/react/components/Stars.tsx)  |
| activeStarContainer | Active star container  |  [Stars](https://github.com/vtex-apps/bazaarvoice/blob/master/react/components/Stars.tsx) |
| inactiveStar | Inactive star  |  [Stars](https://github.com/vtex-apps/bazaarvoice/blob/master/react/components/Stars.tsx) |
| inactiveStarContainer | Inactive star container  |  [Stars](https://github.com/vtex-apps/bazaarvoice/blob/master/react/components/Stars.tsx) |

## Troubleshooting

> You can check if others are passing through similar issues [here](https://github.com/vtex-apps/bazaarvoice/issues). Also feel free to [open issues](https://github.com/vtex-apps/bazaarvoice/issues/new) or contribute with pull requests.

Reviewer 2: ```suggestion
You can check if others are experiencing similar issues [here](https://github.com/vtex-apps/bazaarvoice/issues). Also, feel free to [open issues](https://github.com/vtex-apps/bazaarvoice/issues/new) or contribute with a pull request.
```

## Contributing

> Check it out [how to contribute](https://github.com/vtex-apps/awesome-io#contributing) with this project. 

Reviewer 2: ```suggestion
Take a look at these guidelines: [how to contribute](https://github.com/vtex-apps/awesome-io#contributing).
```

## Tests

To execute our tests go to `react/` folder and run `yarn test`

### Travis CI

[![Build Status](https://travis-ci.org/vtex-apps/menu.svg?branch=master)](https://travis-ci.org/vtex-apps/menu)

## Contributors

Thanks goes to these wonderful people ([emoji key](https://allcontributors.org/docs/en/emoji-key)):

<!-- ALL-CONTRIBUTORS-LIST:START - Do not remove or modify this section -->
<!-- prettier-ignore -->

> <table><tr><td align="center"><a href="http://giovanapereira.com.br/"><img src="https://avatars3.githubusercontent.com/u/26018620?v=4" width="100px;" alt="Giovana Pereira"/><br /><sub><b>Giovana Pereira</b></sub></a><br /><a href="https://github.com/vtex-apps/menu/commits?author=giovanapereira" title="Code">💻</a></td></tr></table>

Reviewer: Remove this
